src/bindings.d.ts

```diff
@@ -1,7 +1,47 @@
-import type { R2Bucket } from '@cloudflare/workers-types'
+import type { Link } from 'multiformats/link'
+import type { Context } from '@web3-storage/gateway-lib'
+import type { CARLink } from 'cardex/api'
+import type { R2Bucket, KVNamespace, RateLimit } from '@cloudflare/workers-types'
+import type { MemoryBudget } from './lib/mem-budget'
+import { CID } from '@web3-storage/gateway-lib/handlers'
+
+export {}
 
 export interface Environment {
+  VERSION: string
   DEBUG: string
   CARPARK: R2Bucket
   CONTENT_CLAIMS_SERVICE_URL?: string
+  RATE_LIMITS_SERVICE_URL?: string
+  ACCOUNTING_SERVICE_URL: string
+  RATE_LIMITER: RateLimit
+  AUTH_TOKEN_METADATA: KVNamespace
+  FF_RATE_LIMITER_ENABLED: boolean
+}
+
+export type GetCIDRequestData = Pick<Request, 'url' | 'headers'>
+
+export type GetCIDRequestOptions = GetCIDRequestData
+
+export interface RateLimitsService {
+  check: (cid: CID, options: GetCIDRequestOptions) => Promise<RateLimitExceeded>
```

Comment:
`options: GetCIDRequestOptions` are they optional? If not then specify as `config` or `context`.